diff --git a/port/atomic_pointer.h b/port/atomic_pointer.h
index 2b485c7..ed7dcff 100644
--- a/port/atomic_pointer.h
+++ b/port/atomic_pointer.h
@@ -36,6 +36,8 @@
 #define ARCH_CPU_X86_FAMILY 1
 #elif defined(__ARMEL__)
 #define ARCH_CPU_ARM_FAMILY 1
+#elif defined(__aarch64__)
+#define ARCH_CPU_ARM64_FAMILY 1
 #endif
 
 namespace leveldb {
@@ -91,6 +93,12 @@ inline void MemoryBarrier() {
 }
 #define LEVELDB_HAVE_MEMORY_BARRIER
 
+#elif defined(ARCH_CPU_ARM64_FAMILY)
+ inline void MemoryBarrier() {
+   asm volatile("dmb sy" : : : "memory");
+ }
+#define LEVELDB_HAVE_MEMORY_BARRIER
+
 #endif
 
 // AtomicPointer built using platform-specific MemoryBarrier()
@@ -147,6 +155,7 @@ class AtomicPointer {
 #undef LEVELDB_HAVE_MEMORY_BARRIER
 #undef ARCH_CPU_X86_FAMILY
 #undef ARCH_CPU_ARM_FAMILY
+#undef ARCH_CPU_ARM64_FAMILY
 
 }  // namespace port
 }  // namespace leveldb
